### PR TITLE
Added isExpired() method #23.

### LINKED ISSUE #23: More flexibility for sessionContext handling
## Steps to reproduce:
1. Try and find out if your session context is expired

## What should happen:
1. There is a way to know if your session has expired without resetting/calling `ensureSesiosActive()`

## What happens:
1. There is no way to achieve such thing 

## Extra info:
- Tested on https://github.com/bunq/sdk_java/tree/0.12.0

## Why
- This way you wont have to execute this code before each call but only if your session did expire

```
apiContext.ensureSessionActive();
apiContext.save();
```
Which means you will be saving an unchanged context on each call

### PR DESCRIPTION
**Pull request by OGKevin**
## Context 
#23 

## What has been done
This pr introduces a new way to handle `SessionContext`. Now it is possible to do the following:

```
    if (apiContext.isExpired()) {
      apiContext.resetSession();
      apiContext.save();
    }
```

This way there is no saving an unchanged context.

Closes #23

### CODE REVIEW

**Comment by OGKevin** [comment]
@dnl-blkv all yours 👀

File: src/main/java/com/bunq/sdk/context/ApiContext.java [hunk 1]
```diff
@@ -230,13 +230,18 @@ private void deleteSession() {
...
+   if (!isSessionActive()){
+     resetSession();
+   }
+  }
+
+  /**
+   * @return True if session has expired, otherwise false.
+   */
+  public boolean isSessionActive() {
+    return !(sessionContext == null)
```
Comment (dnl-blkv) [inline src/main/java/com/bunq/sdk/context/ApiContext.java]: Please use `!=` and order them as:
```
return
    sessionContext != null ||
    getTimeToSessionExpiryInSeconds() > TIME_TO_SESSION_EXPIRY_MINIMUM_SECONDS;
```

File: src/main/java/com/bunq/sdk/context/ApiContext.java [hunk 2]
```diff
@@ -230,13 +230,18 @@ private void deleteSession() {
...
-    if (sessionContext == null) {
-      return;
-    }
+   if (!isSessionActive()){
+     resetSession();
+   }
+  }
+
+  /**
+   * @return True if session has expired, otherwise false.
```
Comment (dnl-blkv) [inline src/main/java/com/bunq/sdk/context/ApiContext.java]: Please remove this comment. This method is self-descriptive enough, and also the comment is not very precise :P

File: src/main/java/com/bunq/sdk/http/ApiClient.java [hunk 3]
```diff
@@ -53,6 +54,18 @@
...
+   * Endpoints not requiring active session for the request to succeed.
+   */
+  private static final String DEVICE_SERVER_URL = "device-server";
+  private static final String INSTALLATION_URL = "installation";
+  private static final String SESSION_SERVER_URL = "session-server";
+  private static List<String> URIS_NOT_REQUIRING_ACTIVE_SESSION = Arrays.asList(
+      DEVICE_SERVER_URL,
+      INSTALLATION_URL,
+      SESSION_SERVER_URL
+      );
```
Comment (dnl-blkv) [inline src/main/java/com/bunq/sdk/http/ApiClient.java]: Is it the formatting that IDE suggested? I mean I would expect there something like:
```
URIS_NOT_REQUIRING_ACTIVE_SESSION = Arrays.asList(
    DEVICE_SERVER_URL,
    INSTALLATION_URL,
    SESSION_SERVER_URL
);
```

Comment (dnl-blkv) [reply src/main/java/com/bunq/sdk/http/ApiClient.java]: Also, please make it `private static` final! `asList` allows for it if I am not mistaken :)

File: src/main/java/com/bunq/sdk/http/ApiClient.java [hunk 4]
```diff
@@ -161,8 +174,10 @@ private URI determineFullUri(String uri, Map<String, String> params) throws URIS
   }
 
   private CloseableHttpResponse executeRequest(HttpUriRequest request,
-      Map<String, String> customHeaders) throws IOException {
-    apiContext.ensureSessionActive();
+      Map<String, String> customHeaders, String uri) throws IOException {
+    if (!URIS_NOT_REQUIRING_ACTIVE_SESSION.contains(uri)) {
+      apiContext.ensureSessionActive();
+    }
```
Comment (dnl-blkv) [inline src/main/java/com/bunq/sdk/http/ApiClient.java]: Please add newline after `if`

**Comment by OGKevin** [comment]
@dnl-blkv all yours again 🙉

**Comment by dnl-blkv** [comment]
@andrederoos

**Pull request merged**